ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/DeploymentsTableContainer.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useQuery } from '@apollo/client';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+
+import useURLSort from 'hooks/useURLSort';
+import useURLPagination from 'hooks/useURLPagination';
+import useURLSearch from 'hooks/useURLSearch';
+import { getHasSearchApplied, getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+import DeploymentsTable, { deploymentListQuery } from '../Tables/DeploymentsTable';
+import TableErrorComponent from '../components/TableErrorComponent';
+import { parseQuerySearchFilter } from '../searchUtils';
+
+const defaultSortFields = ['Deployment', 'Cluster', 'Namespace'];
+
+function DeploymentsTableContainer() {
+    const { searchFilter } = useURLSearch();
+    const querySearchFilter = parseQuerySearchFilter(searchFilter);
+    const isFiltered = getHasSearchApplied(querySearchFilter);
+    const { page, perPage, setPage } = useURLPagination(25);
+    const { sortOption, getSortParams } = useURLSort({
+        sortFields: defaultSortFields,
+        defaultSortOption: {
+            field: 'Deployment',
+            direction: 'asc',
+        },
+        onSort: () => setPage(1),
+    });
+
+    const { error, loading, data } = useQuery(deploymentListQuery, {
+        variables: {
+            query: getRequestQueryStringForSearchFilter({
+                ...querySearchFilter,
+            }),
+            pagination: {
+                offset: (page - 1) * perPage,
+                limit: perPage,
+                sortOption,
+            },
+        },
+    });
+
+    return (
+        <>
+            {loading && (
```

Comment:
On the pages I've been working I've been doing something like this:

```typescript
const { error, loading, data, previousData } = useQuery(...);`
const tableData = data ?? previousData;

{loading && !tableData && (<></>)}
{error && (<></>)}
{tableData && (<></>)}
```

This is because `data` will become `undefined` when the query refetches and the table will be replaced with a loading spinner each time a filter is changed, which feels a little jarring. If the query completed once successfully though we can still show `previousData` until the update occurs.

The downside with this is that stale data is show in the table for a split second, but network lag could cause incorrect data to show for longer.

I think the ideal state to get to is to have some kind of overlay on the table when it is doing a refresh, or reduce the opacity or something that indicates that the data is being updated. I don't think we have a pattern for this currently but it would be interesting to see what other products are doing and propose an idea to the team & UX.

Reply:
i added it to the table for now so at least we have a consistent way to render the tables for now